### DISCUSSION TITLE
React-JUCE for Mobile Android/iOS

### DISCUSSION THREAD

**Discussion by NickolayGerasimenko** [nick-thompson/react-juce#248] [General]
Hi,
We started exploring react-juce to replace our current react-native implementation.
For desktop, all goes pretty well.
But faced a lot of stoppers(C++ compilation errors) trying to build React-JUCE for Android.
We need to support desktop/mobile versions.

What is the current state of React-JUCE for mobile development?

**Comment by nick-thompson** [comment]
Hi @NickolayGerasimenko, sounds great!

> But faced a lot of stoppers(C++ compilation errors) trying to build React-JUCE for Android.
We need to support desktop/mobile versions.

Interesting, can you share some of the build errors?

> What is the current state of React-JUCE for mobile development?

I'll admit that I haven't tried compiling any of my react-juce projects for Android yet, but I have successfully compiled and run my own plugins both on desktop and iOS (both as a standalone and as AUv3). Because it's all JUCE under the hood, we should theoretically be able to offer React-juce mobile support as much as JUCE itself supports mobile.

I hope we can get it there without too much effort as well! I'm interested to take a look at the build errors and see what needs fixing

**Reply by sdeleon28** [reply]
Hey @nick-thompson

According to some [this post in the JUCE forum](https://forum.juce.com/t/using-cmake-flow-to-build-libraries-android-apps/40492/2?u=santiatthmstudio) Android builds aren't supported yet in JUCE CMake (or at least, "we don’t have android or library examples, yet.").

That probably means that react-juce doesn't support Android, at least for now. Am I right?

This is all a bit over my head so I'm not entirely sure.

**Reply by nick-thompson** [reply]
Good question @sdeleon28! I would say that our preferred workflow is through CMake, but the react_juce module can be added as a normal module in the Projucer and used that way, so we should still, in theory, be supporting android that way.

We're aware of a small bug right now if you use the Projucer you must also add a compile definition `-DREACTJUCE_USE_DUKTAPE=1` otherwise you'll hit a bunch of compile errors. We'll be fixing that shortly

@NickolayGerasimenko I'd be curious to know if that fixes the compile errors you were seeing tooo?
